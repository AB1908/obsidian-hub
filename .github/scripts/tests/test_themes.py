import approvaltests
from approvaltests import verify_as_json, verify
from approvaltests.storyboard import StoryBoard

import utils
from tests.test_templates import JINJA_TEMPLATES_DIR
from themes import get_theme_downloads, collect_data_for_theme, ThemeList
from utils import THEMES_JSON_FILE, get_json_from_github


def test_get_theme_downloads() -> None:
    downloads = get_theme_downloads()
    assert len(downloads) >= 117, len(downloads)

    key_of_first_download = list(downloads.keys())[0]
    download = downloads[key_of_first_download]
    assert download["id"] == key_of_first_download, download["id"]
    assert download["download"] > 0, download["download"]

    # TODO Scrub the download numbers
    # TODO Don't add new themes
    # verify_as_json(downloads)


def test_get_community_themes() -> None:
    theme_list: ThemeList = get_json_from_github(THEMES_JSON_FILE)
    assert len(theme_list) >= 115, len(theme_list)

    first_theme = theme_list[0]
    assert len(first_theme["modes"]) > 0
    assert first_theme["author"] != ""
    assert first_theme["name"] != ""
    assert first_theme["screenshot"] != ""

    # TODO Don't add new themes - maybe get a fixed revision
    # verify_as_json(theme_list)


def test_collect_data_for_theme_with_settings() -> None:
    theme_name = "Minimal"
    verify_theme_data(theme_name)


def test_collect_data_for_theme_without_settings() -> None:
    theme_name = "Christmas"
    verify_theme_data(theme_name)


def verify_theme_data(theme_name: str) -> None:
    s = StoryBoard()

    template = utils.get_template_from_directory(JINJA_TEMPLATES_DIR, "theme.md.jinja")
    theme_downloads = get_theme_downloads()

    theme_list: ThemeList = get_json_from_github(THEMES_JSON_FILE)
    theme = None
    for t in theme_list:
        if t["name"] == theme_name:
            theme = t
            break
    assert theme
    assert theme["name"] == theme_name
    assert len(theme["modes"]) > 0
    assert theme["author"] != ""

    s.add_frame(approvaltests.utils.to_json(theme))

    name = collect_data_for_theme(theme, theme_downloads, template)
    assert name == theme_name
    assert theme["user"] != ""
    s.add_frame(approvaltests.utils.to_json(theme))

    # TODO Scrub the download numbers
    # verify(s)
