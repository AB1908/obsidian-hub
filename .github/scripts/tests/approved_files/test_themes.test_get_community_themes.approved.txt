[
    {
        "author": "kognise",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Atom",
        "repo": "kognise/obsidian-atom",
        "screenshot": "screenshot-hybrid.png"
    },
    {
        "author": "cotemaxime",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Amethyst",
        "repo": "cotemaxime/obsidian-amethyst",
        "screenshot": "screenshot.png"
    },
    {
        "author": "insanum",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Obsidian gruvbox",
        "repo": "insanum/obsidian_gruvbox",
        "screenshot": "dark.png"
    },
    {
        "author": "insanum",
        "modes": [
            "dark"
        ],
        "name": "Obsidian Nord",
        "repo": "insanum/obsidian_nord",
        "screenshot": "screen.png"
    },
    {
        "author": "jarodise",
        "modes": [
            "dark"
        ],
        "name": "Dracula for Obsidian",
        "repo": "jarodise/Dracula-for-Obsidian.md",
        "screenshot": "screencap.jpg"
    },
    {
        "author": "lizardmenfromspace",
        "modes": [
            "light"
        ],
        "name": "Gastown",
        "repo": "dogwaddle/obsidian-gastown-theme.md",
        "screenshot": "ObsidianOne.png"
    },
    {
        "author": "deathau",
        "modes": [
            "dark",
            "light"
        ],
        "name": "80s Neon",
        "repo": "deathau/80s-Neon-for-Obsidian.md",
        "screenshot": "screenshot.jpg"
    },
    {
        "author": "deathau",
        "modes": [
            "dark"
        ],
        "name": "Base2Tone",
        "repo": "deathau/Base2Tone-For-Obsidian.md",
        "screenshot": "colours.gif"
    },
    {
        "author": "deathau",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Notation",
        "repo": "deathau/Notation-for-Obsidian",
        "screenshot": "screenshot.jpg"
    },
    {
        "author": "deathau",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Christmas",
        "repo": "deathau/obsidian-christmas-theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "Slowbad",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Solarized",
        "repo": "Slowbad/obsidian-solarized",
        "screenshot": "screenshot.png"
    },
    {
        "author": "ezs",
        "modes": [
            "dark"
        ],
        "name": "Comfort color dark",
        "repo": "obsidian-ezs/obsidian-comfort-color-dark",
        "screenshot": "screencap.png"
    },
    {
        "author": "ezs",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Ursa",
        "repo": "obsidian-ezs/obsidian-ursa",
        "screenshot": "light-theme_full.png"
    },
    {
        "author": "nickmilo",
        "modes": [
            "dark"
        ],
        "name": "Cybertron",
        "repo": "nickmilo/Cybertron",
        "screenshot": "Cybertron.png"
    },
    {
        "author": "karz",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Moonlight",
        "repo": "kartik-karz/moonlight-obsidian",
        "screenshot": "moonlight-theme.png"
    },
    {
        "author": "SeanWcom",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Red Graphite",
        "repo": "seanwcom/Red-Graphite-for-Obsidian",
        "screenshot": "screenshot01.png"
    },
    {
        "author": "karz",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Subtlegold",
        "repo": "kartik-karz/subtlegold-obsidian",
        "screenshot": "subtlegold-theme.png"
    },
    {
        "author": "Sainadh",
        "modes": [
            "light"
        ],
        "name": "Obsidian Boom",
        "repo": "sainadh-d/obsidian-boom",
        "screenshot": "roam-1.png"
    },
    {
        "author": "Reggie",
        "modes": [
            "dark"
        ],
        "name": "Hulk",
        "repo": "pgalliford/Obsidian-theme-Incredible-Hulk",
        "screenshot": "Screen Shot.png"
    },
    {
        "author": "MooddooM",
        "modes": [
            "dark"
        ],
        "name": "Pisum",
        "repo": "GuangluWu/obsidian-pisum",
        "screenshot": "fullpower.png"
    },
    {
        "author": "Boyd",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Traffic Lights",
        "repo": "elliotboyd/obsidian-traffic-lights",
        "screenshot": "dark.png"
    },
    {
        "author": "Mediapathic",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Ars Magna",
        "repo": "mediapathic/obsidian-arsmagna-theme",
        "screenshot": "arsmagna.png"
    },
    {
        "author": "_ph",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Obsdn-Dark-Rmx",
        "repo": "cannibalox/Obsdn-dark-rmx",
        "screenshot": "Obsdn-Dark-Rmx.png"
    },
    {
        "author": "kepano",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Minimal",
        "repo": "kepano/obsidian-minimal",
        "screenshot": "dark-simple.png"
    },
    {
        "author": "rcvd",
        "modes": [
            "dark",
            "light"
        ],
        "name": "obsidian_ia",
        "repo": "rcvd/obsidian_ia",
        "screenshot": "light.png"
    },
    {
        "author": "bernardo_v",
        "modes": [
            "dark"
        ],
        "name": "Charcoal",
        "repo": "bcdavasconcelos/Obsidian-Charcoal",
        "screenshot": "charcoal.png"
    },
    {
        "author": "bernardo_v",
        "modes": [
            "dark"
        ],
        "name": "Panic Mode",
        "repo": "bcdavasconcelos/Obsidian-Panic_Mode",
        "screenshot": "panic.png"
    },
    {
        "author": "bernardo_v",
        "modes": [
            "dark"
        ],
        "name": "Dark Graphite",
        "repo": "bcdavasconcelos/Obsidian-Graphite",
        "screenshot": "graphite.png"
    },
    {
        "author": "bernardo_v",
        "modes": [
            "light"
        ],
        "name": "Ayu",
        "repo": "bcdavasconcelos/Obsidian-Ayu",
        "screenshot": "ayu2.png"
    },
    {
        "author": "bernardo_v",
        "modes": [
            "dark"
        ],
        "name": "Ayu Mirage",
        "repo": "bcdavasconcelos/Obsidian-Ayu_Mirage",
        "screenshot": "ayu1.png"
    },
    {
        "author": "bernardo_v",
        "modes": [
            "light"
        ],
        "name": "GDCT",
        "repo": "bcdavasconcelos/Obsidian-GDCT",
        "screenshot": "gdct.png"
    },
    {
        "author": "bernardo_v",
        "modes": [
            "dark"
        ],
        "name": "GDCT Dark",
        "repo": "bcdavasconcelos/Obsidian-GDCT_Dark",
        "screenshot": "gdct.png"
    },
    {
        "author": "Dubinin Dmitry",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Obuntu",
        "repo": "DubininDmitry/Obuntu-theme-for-Obsidian",
        "screenshot": "screenshot.jpg"
    },
    {
        "author": "_ph",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Ono Sendai",
        "repo": "cannibalox/ono-sendai_obsdn",
        "screenshot": "ono-sendai_obsdn_00.png"
    },
    {
        "author": "Whyl",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Blue Topaz",
        "repo": "whyt-byte/Blue-Topaz_Obsidian-css",
        "screenshot": "preview_Blue Topaz.png"
    },
    {
        "author": "Santi Younger",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Reverie",
        "repo": "santiyounger/Reverie-Obsidian-Theme",
        "screenshot": "img/reverie-2020-09-14-dark.png"
    },
    {
        "author": "kitchenrunner",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Dark Graphite Pie",
        "repo": "ryjjin/Obsidian-Dark-Graphite-Pie-theme",
        "screenshot": "Dark Graphite Pie theme 0.9.4.png"
    },
    {
        "author": "Benny Guo",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Obsidianite",
        "repo": "bennyxguo/Obsidian-Obsidianite",
        "screenshot": "images/demo1.png"
    },
    {
        "author": "Ish Gunacar",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Gitsidian",
        "repo": "ishgunacar/gitsidian",
        "screenshot": "showcase.png"
    },
    {
        "author": "Jamie Brynes",
        "modes": [
            "dark"
        ],
        "name": "Clair de Lune",
        "repo": "jamiebrynes7/clair-de-lune-obsidian-theme",
        "screenshot": "assets/screenshot.png"
    },
    {
        "author": "Manas Sadasivuni",
        "modes": [
            "light"
        ],
        "name": "Molecule",
        "repo": "manassadasivuni/obsidian-molecule",
        "screenshot": "images/Screenshot1.png"
    },
    {
        "author": "Spark",
        "modes": [
            "dark"
        ],
        "name": "Comfort Smooth",
        "repo": "sparklau/comfort-smooth",
        "screenshot": "comfort-smooth.png"
    },
    {
        "author": "dxcore35",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Suddha",
        "repo": "dxcore35/Suddha-theme",
        "screenshot": "Images/Preview1.jpg"
    },
    {
        "author": "mgmeyers",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "California Coast",
        "repo": "mgmeyers/obsidian-california-coast-theme",
        "screenshot": "screenshots/04.png"
    },
    {
        "author": "radekkozak",
        "modes": [
            "dark"
        ],
        "name": "Discordian",
        "repo": "radekkozak/discordian",
        "screenshot": "media/screenshots/discordian-full-mode.png"
    },
    {
        "author": "chad-bennett",
        "modes": [
            "light"
        ],
        "name": "Al Dente",
        "repo": "chad-bennett/al-dente-obsidian-theme",
        "screenshot": "aldente-screenshot.png"
    },
    {
        "author": "Santi Younger",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Wasp",
        "repo": "santiyounger/Wasp-Obsidian-Theme",
        "screenshot": "img/wasp-dark.png"
    },
    {
        "author": "David Golding",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Big Sur Aesthetic",
        "repo": "davidgolding/obsidian-big-sur-aesthetic",
        "screenshot": "screenshot-hybrid.png"
    },
    {
        "author": "lukauskas",
        "branch": "main",
        "modes": [
            "light"
        ],
        "name": "Higlighter",
        "repo": "lukauskas/obsidian-highlighter-theme",
        "screenshot": "screenshots/screenshot-themes-panel.png"
    },
    {
        "author": "SlRvb",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "ITS Theme",
        "repo": "SlRvb/Obsidian--ITS-Theme",
        "screenshot": "ITS.png"
    },
    {
        "author": "Wiktoria Mielcarek",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Spectrum",
        "repo": "Braweria/Spectrum",
        "screenshot": "SpectrumPreview.png"
    },
    {
        "author": "Andrew Lisowski",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Hipstersmoothie",
        "repo": "hipstersmoothie/hipstersmoothie-obsidian-theme",
        "screenshot": "hipstersmoothie-obsidian-theme.png"
    },
    {
        "author": "Hung-Su Nguyen",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Vileplume",
        "repo": "hungsu/vileplume-obsidian",
        "screenshot": "Vileplume.png"
    },
    {
        "author": "Benny Guo",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Aurora",
        "repo": "auroral-ui/aurora-obsidian-md",
        "screenshot": "screenshots/screenshot-1.png"
    },
    {
        "author": "izumin5210",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Iceberg",
        "repo": "izumin5210/obsidian-iceberg",
        "screenshot": "screenshot.png"
    },
    {
        "author": "Ben Hong",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Night Owl",
        "repo": "bencodezen/obsidian-night-owl-theme",
        "screenshot": "obsidian-night-owl-theme.png"
    },
    {
        "author": "Chetachi E.",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Yin and Yang",
        "repo": "chetachiezikeuzor/Yin-and-Yang-Theme",
        "screenshot": "assets/screenshot.png"
    },
    {
        "author": "Mouth on Cloud",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Golden Topaz",
        "repo": "shaggyfeng/obsidian-Golden-Topaz-theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "Mouth on Cloud",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Pink Topaz",
        "repo": "shaggyfeng/obsidian-Pink-topaz-theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "sergey",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Dark Moss",
        "repo": "sergey900553/obsidian_githublike_theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "Witt Allen",
        "modes": [
            "dark"
        ],
        "name": "Mammoth",
        "repo": "Wittionary/mammoth-obsidian-theme",
        "screenshot": "screenshots/fullscreen-formatting-examples.png"
    },
    {
        "author": "Luke Ruokaismaki",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Rmaki",
        "repo": "luke-rmaki/rmaki-obsidian",
        "screenshot": "screenshot.png"
    },
    {
        "author": "ArtexJay",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Cyber Glow",
        "repo": "ArtexJay/Obsidian-CyberGlow",
        "screenshot": "Screenshot.png"
    },
    {
        "author": "johackim",
        "modes": [
            "dark"
        ],
        "name": "Darkyan",
        "repo": "johackim/obsidian-darkyan",
        "screenshot": "screenshot.png"
    },
    {
        "author": "MrGlitchByte",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Everforest",
        "repo": "mrglitchbyte/obsidian_everforest",
        "screenshot": "dark_v2.png"
    },
    {
        "author": "Zaher Al Majed",
        "branch": "main",
        "modes": [
            "light"
        ],
        "name": "Illusion",
        "repo": "ZaherAlMajed/Illusion-Theme.md",
        "screenshot": "screenshot.png"
    },
    {
        "author": "Tyler Nguyen",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Horizon",
        "repo": "tylernguyen/obsidian-horizon",
        "screenshot": "assets/preview.png"
    },
    {
        "author": "Ruslan Gagushin",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Tokyo Night",
        "repo": "RuslanGagushin/Tokyo-Night-Obsidian-Theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "nikbrunner",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Deep Work",
        "repo": "nikbrunner/obsidian-deep-work-theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "Ivan Chernov",
        "modes": [
            "dark"
        ],
        "name": "Blackbird",
        "repo": "vanadium23/obsidian-blackbird-theme",
        "screenshot": "images/example.png"
    },
    {
        "author": "Kinmury Ditamir",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Golden Coffee",
        "repo": "kinmury/GoldenCoffeeTheme",
        "screenshot": "Screenshots/Showcase.png"
    },
    {
        "author": "Chrismettal",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Behave dark",
        "repo": "Chrismettal/Obsidian-Behave-dark",
        "screenshot": "Screenshot.png"
    },
    {
        "author": "SMUsamaShah",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Obsidian Windows 98 Edition",
        "repo": "SMUsamaShah/Obsidian-Win98-Edition",
        "screenshot": "screenshots/main.png"
    },
    {
        "author": "lizardmenfromspace",
        "branch": "main",
        "modes": [
            "light"
        ],
        "name": "Lizardmen Zettelkasten",
        "repo": "dogwaddle/lizardmen-zettelkasten",
        "screenshot": "screenshot.png"
    },
    {
        "author": "s\u00f3 erick mesmo",
        "branch": "main",
        "modes": [
            "light"
        ],
        "name": "Cidreira Coast",
        "repo": "firinael/obsidian-cidreira-coast-theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "pseudometa",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Shimmering Focus",
        "repo": "chrisgrieser/shimmering-focus",
        "screenshot": "docs/images/Promo%20Screenshot/promo-screenshot.png"
    },
    {
        "author": "Ali Soueidan",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Firefly",
        "repo": "lazercaveman/obsidian-firefly-theme",
        "screenshot": "firefly-theme-screenshot.png"
    },
    {
        "author": "cybeerboy",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Deep",
        "repo": "cybeerboy/Deep-for-Obsidian",
        "screenshot": "screenshot.png"
    },
    {
        "author": "zacharyc",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Purple Owl",
        "repo": "zacharyc/purple-owl-theme",
        "screenshot": "purple-owl-theme.png"
    },
    {
        "author": "Piglet1236",
        "modes": [
            "dark"
        ],
        "name": "Emerald",
        "repo": "gracejoseph1236/obsidian-emerald",
        "screenshot": "example.png"
    },
    {
        "author": "tomzorz",
        "modes": [
            "dark"
        ],
        "name": "Sodalite",
        "repo": "tomzorz/Sodalite",
        "screenshot": "screenshot.png"
    },
    {
        "author": "Seraaron",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Agora",
        "repo": "Seraaron/agora-obsidian-theme",
        "screenshot": "agora-theme-preview.png"
    },
    {
        "author": "Josh Kasap",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Faded",
        "repo": "JoshKasap/Obsidian-Faded-Theme",
        "screenshot": "Faded.png"
    },
    {
        "author": "Marcus Olsson",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Creature",
        "repo": "marcusolsson/obsidian-creature-theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "jdanielmourao",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Sanctum",
        "repo": "jdanielmourao/obsidian-sanctum",
        "screenshot": "cover.png"
    },
    {
        "author": "Zarkov",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Slytherin",
        "repo": "MatheusZarkov/Obsidian-Slytherin",
        "screenshot": "Cover.png"
    },
    {
        "author": "cassidoo",
        "branch": "main",
        "modes": [
            "light"
        ],
        "name": "Cardstock",
        "repo": "cassidoo/cardstock",
        "screenshot": "screenshot.png"
    },
    {
        "author": "Nilahn",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Pine Forest Berry",
        "repo": "Nilahn/pine_forest_berry",
        "screenshot": "Screenshot PFB 1.png"
    },
    {
        "author": "mimishahzad",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Ros\u00e9 Pine Moon",
        "repo": "mimishahzad/rose-pine-moon-obsidian",
        "screenshot": "assets/template.png"
    },
    {
        "author": "Piglet1236",
        "modes": [
            "dark"
        ],
        "name": "Ruby",
        "repo": "gracejoseph1236/obsidian-ruby",
        "screenshot": "example.png"
    },
    {
        "author": "Damian Korcz",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Prism",
        "repo": "damiankorcz/Prism-Theme",
        "screenshot": "assets/screenshots/Community Themes Screenshot.png"
    },
    {
        "author": "Kinmury Ditamir",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Golden Book",
        "repo": "kinmury/Golden-Book",
        "screenshot": "Gallery/Gallery1.png"
    },
    {
        "author": "Piglet1236",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Carnelian",
        "repo": "gracejoseph1236/obsidian-carnelian",
        "screenshot": "example.png"
    },
    {
        "author": "Joschua",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Obsidian You",
        "repo": "selfire1/obsidian-you-theme",
        "screenshot": "theme-picker.png"
    },
    {
        "author": "Erin Schnabel",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Ebullientworks",
        "repo": "ebullient/obsidian-theme-ebullientworks",
        "screenshot": "images/ebullientworks-theme.jpg"
    },
    {
        "author": "s\u00f3 erick mesmo",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Friendly",
        "repo": "firinael/obsidian-friendly-theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "<Serice",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Purple Aurora",
        "repo": "AndreasStandar/Obsidian-Theme---Purple-Aurora",
        "screenshot": "1 Purple Aurora screenshot.png"
    },
    {
        "author": "Cecilia May",
        "branch": "main",
        "modes": [
            "light",
            "dark"
        ],
        "name": "Primary",
        "repo": "ceciliamay/obsidianmd-theme-primary",
        "screenshot": "assets/primary-heroimg.png"
    },
    {
        "author": "zcysxy",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Terminal",
        "repo": "zcysxy/Obsidian-Terminal-Theme",
        "screenshot": "screenshots/terminal.png"
    },
    {
        "author": "Emrie Candera",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Bubble Space",
        "repo": "Emrie-Candera/Bubble-Space-Theme",
        "screenshot": "screenshot.png"
    },
    {
        "author": "Colin Eckert",
        "branch": "main",
        "modes": [
            "light",
            "dark"
        ],
        "name": "Things",
        "repo": "colineckert/obsidian-things",
        "screenshot": "assets/main-demo.png"
    },
    {
        "author": "Hung-Su Nguyen",
        "branch": "main",
        "modes": [
            "light",
            "dark"
        ],
        "name": "Typomagical",
        "repo": "hungsu/typomagical-obsidian",
        "screenshot": "Typomagical-split.jpg"
    },
    {
        "author": "Curio Heart",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Wyrd",
        "repo": "curio-heart/obsidian-wyrd",
        "screenshot": "img/Wyrd.png"
    },
    {
        "author": "crashmoney",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Typewriter",
        "repo": "crashmoney/obsidian-typewriter",
        "screenshot": "cover.jpg"
    },
    {
        "author": "MattWilcox",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "WilcoxOne",
        "repo": "MattWilcox/obsidian-wilcox-one",
        "screenshot": "wilcoxonedark.png"
    },
    {
        "author": "Kinmury Ditamir",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Alchemy",
        "repo": "kinmury/Alchemy",
        "screenshot": "Showcase/ShowcaseF.png"
    },
    {
        "author": "Thiews",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Harmonic",
        "repo": "Thiews/Obsidian-Harmonic",
        "screenshot": "cover.png"
    },
    {
        "author": "Zeno Rocha & Andy Byers",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Dracula Official",
        "repo": "dracula/obsidian",
        "screenshot": "screenshot.png"
    },
    {
        "author": "dlccyes",
        "modes": [
            "dark"
        ],
        "name": "Nebula",
        "repo": "dlccyes/obsidian-nebula",
        "screenshot": "showcase1.png"
    },
    {
        "author": "AidanGlickman",
        "modes": [
            "dark"
        ],
        "name": "Obsidian Nord Enhanced",
        "repo": "aidanglickman/obsidian-nord-enhanced",
        "screenshot": "_media/scrot.png"
    },
    {
        "author": "bLaCkwEw",
        "branch": "master",
        "modes": [
            "dark"
        ],
        "name": "Dracula Slim",
        "repo": "bLaCkwEw/Dracula-Slim",
        "screenshot": "screenshot.png"
    },
    {
        "author": "jaysan",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Sandstorm",
        "repo": "jaysan0/obsidian-sandstorm",
        "screenshot": "screenshot.png"
    },
    {
        "author": "nickmilo",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "LYT Mode",
        "repo": "nickmilo/LYT-Mode",
        "screenshot": "lyt-mode-graphic-1.jpg"
    },
    {
        "author": "Alex B",
        "branch": "main",
        "modes": [
            "dark"
        ],
        "name": "Carpe Noctem",
        "repo": "operator-axel/obsdian_theme--Carpe_Noctem",
        "screenshot": "screenshot.png"
    },
    {
        "author": "natelandau",
        "branch": "main",
        "modes": [
            "dark",
            "light"
        ],
        "name": "Vintage Theme",
        "repo": "natelandau/obsidian-vintage-theme",
        "screenshot": "_media/light.png"
    }
]
